### PR TITLE
Bump 2.7 apiserver version

### PR DESCRIPTION
**Pull request by JonCrowther**
## Problem
Currently rancher does not have the latest 2.7 version of `github.com/rancher/apiserver`
 
## Solution
Upgrade the go.mod via `go get -u github.com/rancher/apiserver@d134152ce6fb` and `go mod tidy`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -146,20 +146,20 @@ require (
 	helm.sh/helm/v3 v3.9.0
 	k8s.io/api v0.26.1
 	k8s.io/apiextensions-apiserver v0.25.12
-	k8s.io/apimachinery v0.26.1
-	k8s.io/apiserver v0.25.12
+	k8s.io/apimachinery v0.28.3
+	k8s.io/apiserver v0.28.3
```
Comment (maxsokolovsky) [inline go.mod:150]: Just to be sure - already 1.28, not 1.27? In v2.8, we still use 1.27...

Comment (JonCrowther) [reply go.mod]: Good catch. I re-did the work in a less intrusive way.

**Review by tomleb — APPROVED** [review]
LGTM, does this need some tests/QA?

**Review by maxsokolovsky — APPROVED** [review]

**Comment by JonCrowther** [comment]
> LGTM, does this need some tests/QA?

Since it's just a dependency upgrade, I believe the existing CI is adequate

**Pull request merged**